src/brod_producer_buffer.erl

```diff
@@ -34,26 +36,26 @@
 
 %% keep data in fun() to avoid huge log dumps in case of crash etc.
 -type data() :: fun(() -> kafka_kv()).
--type call() :: brod_call_ref().
 
 -record(req,
-        { call :: call()
-        , data :: data()
+        { call_ref :: brod_call_ref()
+        , data     :: data()
+        , bytes    :: non_neg_integer()
```

Comment:
would it make moe sense to have a failure_counter here?
the counter is bumped whenever SenFun returned error, or when it is 'nack'-ed.

then SendFun do an exit(LastError) (or return a non-retriable error Reason) when failure_count hits the limit.

comparing to a global counter in the producer looping state, this approaches is more robust as there is no need to reset the counter etc.